src/inmanta/server/services/environment_metrics_service.py

```
@@ -0,0 +1,120 @@
+        """
+        Invoked by the `EnvironmentMetricsService` at the end of the metrics collection interval.
+        Returns the metrics collected by this MetricCollector within the past metrics collection interval.
+
+        The *_interval arguments are present, because this method is intended to perform a query on
+        the database. No in-memory state is being stored by this metrics collector. The provided interval
+        should be interpreted as [start_interval, end_interval[
+
+        :param start_interval: The start time of the metrics collection interval (inclusive).
+        :param end_interval: The end time of the metrics collection interval (exclusive).
+        :result: The metrics collected by this MetricCollector within the past metrics collection interval.
+        """
+        raise NotImplementedError()
+
+
+class EnvironmentMetricsService(protocol.ServerSlice):
+    """Slice for the management of metrics"""
+
+    def __init__(self) -> None:
+        super(EnvironmentMetricsService, self).__init__(SLICE_ENVIRONMENT_METRICS)
+        self.metrics_collectors: List[MetricsCollector] = []
+
+    def get_dependencies(self) -> List[str]:
+        return [SLICE_DATABASE]
+
+    def get_depended_by(self) -> List[str]:
+        return [SLICE_TRANSPORT]
+
+    async def start(self) -> None:
+        await super().start()
+        self.schedule(self.flush_metrics, 60, initial_delay=0, cancel_on_stop=True)
+
+    def register_metric_collector(self, metrics_collector: MetricsCollector) -> None:
+        """
+        Register the given metrics_collector.
+        """
+        if not any(metric.get_metric_name() == metrics_collector.get_metric_name() for metric in self.metrics_collectors):
+            self.metrics_collectors.append(metrics_collector)
+        else:
+            LOGGER.warning(f"There already is a metric collector with the name {metrics_collector.get_metric_name()}")
```

This case should never happen. Let's raise an exception here instead.